pkg/applicationmanager/controllers/applicationrestore.go

```diff
@@ -149,14 +149,33 @@ func (a *ApplicationRestoreController) createNamespaces(backup *storkapi.Applica
 			log.ApplicationRestoreLog(restore).Infof("Creating dest namespace %v", ns.Name)
 			if err != nil {
 				if errors.IsAlreadyExists(err) {
+					oldNS, err := core.Instance().GetNamespace(ns.GetName())
+					if err != nil {
+						return err
+					}
+					annotations := make(map[string]string)
+					labels := make(map[string]string)
+					if restore.Spec.ReplacePolicy == storkapi.ApplicationRestoreReplacePolicyDelete {
+						// overwrite all annotation in case of replace policy set to delete
+						annotations = ns.GetAnnotations()
+						labels = ns.GetLabels()
+					} else if restore.Spec.ReplacePolicy == storkapi.ApplicationRestoreReplacePolicyRetain {
+						// merge annotation in case of replace policy is set to retain
+						annotations = oldNS.GetAnnotations()
+						for k, v := range ns.GetAnnotations() {
```

Comment:
So you are overwriting existing annotations with the new ones? You want to do the opposite. Same for labels

Reply:
make sense, updated